test/e2e_node/cpu_manager_test.go

```diff
@@ -1172,6 +1303,75 @@ func isSMTAlignmentError(pod *v1.Pod) bool {
 	return re.MatchString(pod.Status.Reason)
 }
 
+// getNumaNodeCPUs retrieves CPUs for each NUMA node.
+func getNumaNodeCPUs() (map[int]cpuset.CPUSet, error) {
+	numaNodes := make(map[int]cpuset.CPUSet)
+	nodePaths, err := filepath.Glob("/sys/devices/system/node/node*/cpulist")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, nodePath := range nodePaths {
+		data, err := os.ReadFile(nodePath)
+		framework.ExpectNoError(err, "Error obtaning CPU information from the node")
+		cpuSet := strings.TrimSpace(string(data))
+		cpus, err := cpuset.Parse(cpuSet)
+		framework.ExpectNoError(err, "Error parsing CPUset")
+
+		// Extract node ID from path (e.g., "node0" -> 0)
+		base := filepath.Base(filepath.Dir(nodePath))
+		nodeID, err := strconv.Atoi(strings.TrimPrefix(base, "node"))
+		if err != nil {
+			continue
+		}
+		numaNodes[nodeID] = cpus
+	}
+
+	return numaNodes, nil
+}
+
+// computeNUMADistribution calculates CPU distribution per NUMA node.
+func computeNUMADistribution(allocatedCPUs cpuset.CPUSet) map[int]int {
```

Comment:
if you resubmit anyway: what about adding a couple logs to troubleshoot failures?
```
// computeNUMADistribution calculates CPU distribution per NUMA node.
func computeNUMADistribution(allocatedCPUs cpuset.CPUSet) map[int]int {
        numaCPUs, err := getNumaNodeCPUs()
        framework.ExpectNoError(err, "Error retrieving NUMA nodes")
        framework.Logf("NUMA Node CPUs allocation: %v", numaCPUs)

        distribution := make(map[int]int)
        for node, cpus := range numaCPUs {
                distribution[node] = cpus.Intersection(allocatedCPUs).Size()
        }
        framework.Logf("allocated CPUs %s distribution: %v", allocatedCPUs.String(), distribution)
        return distribution
}
```